gdc_ng_models/alembic/versions/38eab5dd7b0d_extend_entity_sets_with_intent_and_ttl.py

```diff
@@ -0,0 +1,50 @@
+"""extend entity_sets with intent and ttl
+
+Revision ID: 38eab5dd7b0d
+Revises: 12dbbcac7a1d
+Create Date: 2024-04-12 15:51:16.241332
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+from gdc_ng_models.models import entity_set
+
+# revision identifiers, used by Alembic.
+revision = "38eab5dd7b0d"  # pragma: allowlist secret
+down_revision = "12dbbcac7a1d"  # pragma: allowlist secret
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    intent_type = postgresql.ENUM(
+        "unknown", "internal", "external", "portal", "user", name="intent_type"
+    )
+    intent_type.create(op.get_bind(), checkfirst=True)
+    op.add_column(
+        "entity_set",
+        sa.Column(
+            "intent_type",
+            intent_type,
+            server_default="unknown",
```

Comment:
Is it possible to reference the unknown enum rather than a string literal?

Reply:
I switched everything to literal values. @see the catch-22 explanation.